Doc/library/abc.rst

```diff
@@ -110,55 +109,61 @@ a helper class :class:`ABC` to alternatively define ABCs through inheritance:
 
       .. XXX explain the "usual mechanism"
 
-
-   For a demonstration of these concepts, look at this example ABC definition::
-
-      class Foo:
-          def __getitem__(self, index):
-              ...
-          def __len__(self):
-              ...
-          def get_iterator(self):
-              return iter(self)
-
-      class MyIterable(ABC):
-
-          @abstractmethod
-          def __iter__(self):
-              while False:
-                  yield None
-
-          def get_iterator(self):
-              return self.__iter__()
-
-          @classmethod
-          def __subclasshook__(cls, C):
-              if cls is MyIterable:
-                  if any("__iter__" in B.__dict__ for B in C.__mro__):
-                      return True
-              return NotImplemented
-
-      MyIterable.register(Foo)
-
-   The ABC ``MyIterable`` defines the standard iterable method,
-   :meth:`~iterator.__iter__`, as an abstract method.  The implementation given
-   here can still be called from subclasses.  The :meth:`get_iterator` method
-   is also part of the ``MyIterable`` abstract base class, but it does not have
-   to be overridden in non-abstract derived classes.
-
-   The :meth:`__subclasshook__` class method defined here says that any class
-   that has an :meth:`~iterator.__iter__` method in its
-   :attr:`~object.__dict__` (or in that of one of its base classes, accessed
-   via the :attr:`~class.__mro__` list) is considered a ``MyIterable`` too.
-
-   Finally, the last line makes ``Foo`` a virtual subclass of ``MyIterable``,
-   even though it does not define an :meth:`~iterator.__iter__` method (it uses
-   the old-style iterable protocol, defined in terms of :meth:`__len__` and
-   :meth:`__getitem__`).  Note that this will not make ``get_iterator``
-   available as a method of ``Foo``, so it is provided separately.
-
-
-
+      **Example use**::
+      
+         from abc import ABC, abstractmethod
+
+         class MyIterable(ABC):
+            @abstractmethod
+            def __iter__(self):
+               raise NotImplementedError
+            
+            @classmethod
+            def __subclasshook__(cls, subclass):
+               """Check if ``subclass`` is a virtual subclass of ``MyIterable``."""
+
+               # This behaviour shouldn't be heritable.
+               if cls is MyIterable:
+                   return hasattr(subclass, "__iter__")
+               
+               # Defer back to the issubclass algorithm to determine.
+               return NotImplemented
+      
+      ::
+
+         >>> MyIterable in list.__bases__  # Does `list` inherit from `MyIterable`?
```

Comment:
I think that you version has completely change the meaning of this example. As far as I understand, it is used to show both `__subclasshook__` and `.register`

In your PR the second part is gone.

Reply:
That's true, the meaning is changed in this PR. I don't think we need to keep `.register` in the subclasshook example. For one it has its own full example right above[1] in the docs.

This change makes the subclasshook docs more specific and removes the stuff about old style iteration etc. that I would argue detract from the existing docs. 

1.https://docs.python.org/3.12/library/abc.html#abc.ABCMeta.register